src/node.cc

```diff
@@ -4244,11 +4249,15 @@ static void StartNodeInstance(void* arg) {
       SealHandleScope seal(isolate);
       bool more;
       do {
+#ifndef NODE_EMBEDDED_MODE
         v8::platform::PumpMessageLoop(default_platform, isolate);
+#endif  // NODE_EMBEDDED_MODE
         more = uv_run(env->event_loop(), UV_RUN_ONCE);
 
         if (more == false) {
+#ifndef NODE_EMBEDDED_MODE
           v8::platform::PumpMessageLoop(default_platform, isolate);
```

Comment:
If embedders don't use it, would it make sense to break it out into a separate compilation unit that's skipped for embedded builds?  I'm okay in principle with supporting embedded builds but this #ifdef soup isn't very pleasant.